compiler/lib/effects.ml

```diff
@@ -139,6 +139,28 @@ let dominance_frontier g idom =
     g.preds;
   frontiers
 
+(* Last instruction of a block, ignoring events *)
+let rec last_instr l =
+  match l with
+  | [] -> None
+  | [ i ] | [ i; (Event _, _) ] -> Some i
```

Comment:
This works because we avoid consecutive events. Maybe this invariant should be added to code.ml